p2p/errors.go

```diff
@@ -172,3 +172,13 @@ type ErrNetAddressLookup struct {
 func (e ErrNetAddressLookup) Error() string {
 	return fmt.Sprintf("Error looking up host (%s): %v", e.Addr, e.Err)
 }
+
+// ErrDialingOrExistingAddress indicates that we're currently
+// dialing this address or it belongs to an existing peer.
+type ErrDialingOrExistingAddress struct {
```

Comment:
Nit: maybe rename to `ErrIsDialingOrExistingAddress` or `ErrCurrentlyDialingOrExistingAddress`? When I first read the code I thought this mean "error while dialing or dialing an existing address".